x-pack/plugins/enterprise_search/public/applications/enterprise_search_content/components/search_index/pipelines/ml_inference/ml_inference_logic.ts

```diff
@@ -93,11 +93,25 @@ export const EMPTY_PIPELINE_CONFIGURATION: InferencePipelineConfiguration = {
   modelID: '',
   pipelineName: '',
   sourceField: '',
+  targetField: '',
 };
 
 const API_REQUEST_COMPLETE_STATUSES = [Status.SUCCESS, Status.ERROR];
 const DEFAULT_CONNECTOR_FIELDS = ['body', 'title', 'id', 'type', 'url'];
 
+const getFullTargetFieldName = (
+  sourceField: string,
+  targetField: string | undefined,
+  isTextExpansionModelSelected: boolean
+) => {
+  let suffixedTargetField = (targetField ?? '').length > 0 ? targetField! : sourceField;
```

Comment:
suggestion: you can condense this and the if statement into: 
```
const suffixedTargetField = `${targetField || sourceField}${isTextExpansionModelSelected ? '_expanded' : ''`;
```

Also as a sidenote, we should never use the `!` override.

Reply:
Yeah I get how you landed on `!`. The issue is that this works now, but if we change this code in the future there's a chance it suppresses valid type errors that create in-production issues. I've seen this kind of thing happen many times. Rather than using `!` we should take the time to make TS recognize what's happening (eg assigning to an intermediate variable) or rewriting the thing.